core/workflow/rolebase.go

```diff
@@ -56,12 +57,42 @@ type roleBase struct {
 	status      SafeStatus
 	state       SafeState
 
-	Defaults *gera.StringWrapMap `yaml:"defaults,omitempty"`
-	Vars     *gera.StringWrapMap `yaml:"vars,omitempty"`
-	UserVars *gera.StringWrapMap `yaml:"-"`
-	Locals   map[string]string   `yaml:"-"` // only used for passing iterator from template to new role
-	Bind     []channel.Inbound   `yaml:"bind,omitempty"`
-	Enabled  string              `yaml:"enabled,omitempty"`
+	Defaults *gera.WrapMap[string, string] `yaml:"defaults,omitempty"`
+	Vars     *gera.WrapMap[string, string] `yaml:"vars,omitempty"`
+	UserVars *gera.WrapMap[string, string] `yaml:"-"`
+	Locals   map[string]string             `yaml:"-"` // only used for passing iterator from template to new role
+	Bind     []channel.Inbound             `yaml:"bind,omitempty"`
+	Enabled  string                        `yaml:"enabled,omitempty"`
+}
+
+func kvStoreUnmarshalYAMLWithTags(w gera.Map[string, string], unmarshal func(interface{}) error) error {
```

Comment:
shouldn't this function be also tested?

Reply:
We can indeed test it, but a test probably belongs in the `workflow` package test and not in the one for the `gera` package.